examples/lighting-app/esp32/main/main.cpp

```diff
@@ -86,7 +90,12 @@ static void InitServer(intptr_t context)
     chip::Server::GetInstance().Init(initParams);
 
     // Initialize device attestation config
+#if CONFIG_ENABLE_ESP32_FACTORY_DATA_PROVIDER
+    SetDeviceAttestationCredentialsProvider(&ESP32FactoryDataProvider::GetInstance());
```

Comment:
You don't need a `GetInstance` here. You can have an actual object instance globally held in the anonymous namespace, since ESP32FactoryDataProvider is a concrete class. It does not need to be a singleton with getter at all.

Reply:
Using the global instance of the class.